lib/identity_cache/tracking.rb

```diff
@@ -0,0 +1,70 @@
+# frozen_string_literal: true
+
+module IdentityCache
+  module Tracking
+    TrackedObject = Struct.new(:object, :includes, :caller, :accessed_associations)
+
+    extend self
+
+    def tracked_objects
+      Thread.current[:idc_tracked_objects] ||= {}
+    end
+
+    def reset_tracked_objects
+      Thread.current[:idc_tracked_objects] = {}
+    end
+
+    def track_object(object, includes)
+      return unless object_tracking_enabled
+      locations = caller(1, 20)
+      tracked_objects[object] = TrackedObject.new(object, Array(includes), locations, Set.new)
```

Comment:
Object deserialization is already a hotspot.  So this could add a lot of instrumentation overhead.  I'm guessing the plan is to only use this for a sampling of requests, right?

Reply:
Yeah, I think it's only feasible to use this on a tiny percentage of requests.

Reply:
Looks like the PR description already answered my question.